bundles/io/org.openhab.io.gcal/src/main/java/org/openhab/io/gcal/internal/GCalEventDownloader.java

```diff
@@ -135,69 +190,84 @@ protected void execute() {
+    public static Events downloadEventFeed() {
         // TODO: teichsta: there could be more than one calender url in openHAB.cfg
         // for now we accept this limitation of downloading just one feed ...
 
-        if (StringUtils.isBlank(url)) {
-            logger.warn("Login aborted no url");
-            return null;
-        }
-
-        if (StringUtils.isBlank(username) && StringUtils.isBlank(password)) {
-            logger.info("gcal without username and password (make sure url is accessable without those)");
-        } else if (!StringUtils.isBlank(username) && !StringUtils.isBlank(password)) {
-            logger.info("gcal with username and password");
-        } else {
-            logger.warn("Login aborted none of the 2 cased are fulfilled 1)url, username, password 2) url ");
+        if (StringUtils.isBlank(calendar_name)) {
+            logger.warn("Login aborted no calendar name defined");
             return null;
         }
+        // authorization
+        Credential credential = getCredential();
+
+        // set up global Calendar instance
+        com.google.api.services.calendar.Calendar client = new com.google.api.services.calendar.Calendar.Builder(
+                HTTP_TRANSPORT, JSON_FACTORY, credential).setApplicationName("openHAB").build();
+
+        String calendarID = null;
+
+        if (!calendar_name.equals("primary")) {
+            CalendarList calfeed = null;
+            try {
+                calfeed = client.calendarList().list().execute();
+            } catch (com.google.api.client.auth.oauth2.TokenResponseException ae) {
+                logger.error("authentication failed: {}", ae.getMessage());
+            } catch (IOException e1) {
+                logger.error("authentication I/O exception: {}", e1.getMessage());
+            }
 
-        try {
-            URL feedUrl = new URL(url);
+            if (calfeed != null && calfeed.getItems() != null) {
+                for (CalendarListEntry entry : calfeed.getItems()) {
+                    if (entry.getSummary().equals(calendar_name)) {
+                        calendarID = entry.getId();
+                        logger.debug("Got calendar {} CalendarID: {}", calendar_name, calendarID);
+                    }
+                }
+            }
 
-            CalendarService myService = new CalendarService("openHAB");
-            if (!StringUtils.isBlank(username) && !StringUtils.isBlank(password)) {
-                myService.setUserCredentials(username, password);
+            if (calendarID == null) {
+                logger.error("Calendar {} not found", calendar_name);
+                return null;
             }
+        } else {
+            calendarID = calendar_name;
+        }
 
-            CalendarQuery myQuery = new CalendarQuery(feedUrl);
-            DateTime start = DateTime.now();
-            DateTime end = new DateTime(DateTime.now().getValue() + (2 * refreshInterval));
+        DateTime start = new DateTime(new Date());
```

Comment:
My (minor) point was that `new DateTime()` is identical in result to `new DateTime(new Date())`.